modules/format/src/updating/merge-update-operations.ts

```diff
@@ -130,6 +133,55 @@ function mergeMulOperand<OP extends "$mul">(
   return ret;
 }
 
+function mergePullOperand<OP extends "$pull">(
+  operand1: RegularUpdateOperand<OP>,
+  operand2: RegularUpdateOperand<OP>
+): RegularUpdateOperand<OP> {
+  const ret: RegularUpdateOperand<OP> = Object.assign({}, operand1);
+  Object.entries(operand2).forEach(([k, v]) => {
+    if (k in ret) {
+      const query1 = ret[k];
+      const query2 = v;
+      if ("$eq" in query1 && "$eq" in query2) {
+        // case: {anyPath: $eq: anyValue} + {anyPath: $eq: anyValue}
+        const value1 = query1["$eq"];
+        const value2 = query2["$eq"];
+        if (value1 == value2) {
+          ret[k] = v;
+        } else {
+          ret[k] = { $in: [query1["$eq"], query2["$eq"]] };
+        }
+      } else if ("$in" in query1 && "$eq" in query2) {
+        // case: {anyPath: $in: [anyValues]} + {anyPath: $eq: anyValue}
+        const values = query1["$in"] as any[];
+        const value = query2["$eq"];
+        if (values.includes(value)) {
+          ret[k] = query1;
+        } else {
+          ret[k] = { $in: [...values, value] };
+        }
+      } else if ("$in" in query2 && "$eq" in query1) {
+        // case: {anyPath: $eq: anyValue} + {anyPath: $in: [anyValues]}
+        const values = query2["$in"] as any[];
+        const value = query1["$eq"];
+        if (values.includes(value)) {
+          ret[k] = query2;
+        } else {
+          ret[k] = { $in: [value, ...values] };
+        }
+      } else if ("$in" in query1 && "$in" in query2) {
+        // case: {anyPath: $in: [anyValues]} + {anyPath: $in: [anyvalues]}
+        const values1 = query1["$in"] as any[];
+        const values2 = query2["$in"] as any[];
+        ret[k] = { $in: Array.from(new Set(values1.concat(values2))) };
```

Comment:
Please add an explanation comment for this line. Not all engineers can understand the purpose of this code (make the array unique).